pkg/gofr/datasource/kv-store/badger/badger.go

```diff
@@ -0,0 +1,125 @@
+package badger
+
+import (
+	"context"
+	"errors"
+
+	"github.com/dgraph-io/badger/v4"
+)
+
+type Configs struct {
+	DirPath string
+}
+
+type client struct {
+	db      *badger.DB
+	configs Configs
+	logger  Logger
+	metrics Metrics
+}
+
+func New(configs Configs) *client {
+	return &client{configs: configs}
+}
+
+// UseLogger sets the logger for the MongoDB client which asserts the Logger interface.
```

Comment:
I think there is a copy pasta 

```suggestion
// UseLogger sets the logger for the badgerDB client which asserts the Logger interface.
```